src/nifreeze/model/dmri.py

```diff
@@ -173,7 +175,15 @@ class AverageDWIModel(ExpectationModel):
 
     __slots__ = ("_th_low", "_th_high", "_detrend")
 
-    def __init__(self, dataset, stat="median", th_low=100, th_high=100, detrend=False, **kwargs):
+    def __init__(
+        self,
+        dataset: DWI,
+        stat: str = "median",
+        th_low: numbers.Number = 100,
+        th_high: numbers.Number = 100,
```

Comment:
We should not bother with `numbers`: https://github.com/python/mypy/issues/3186

```suggestion
        th_low: float = 100,
        th_high: float = 100,
```